selection/randomized/query.py

```diff
@@ -1838,3 +1863,117 @@ def _weights(self, candidate):
         return candidate_sample, np.exp(_logratio)
 
 
+class opt_weighted_intervals(object): # intervals_from_sample):
+
+    """
+    Location family based intervals... (cryptic)
+    randomization density should be `g` composed with the affine
+    mapping and take an argument like one row of sample
+    target_linear is the linear part of the affine mapping with
+    respect to target
+    weights for a given candidate will look like
+          randomization_density(sample + (candidate, 0, 0) - (reference, 0, 0)) /
+          randomization_density(sample)
+    if the samples are samples of \bar{\beta}. if we have samples of
+    \Delta from our reference, then the weights will look like
+    randomization_density(sample + (candidate, 0, 0))
+    randomization_density(sample + (reference, 0, 0))
+    WE ARE ASSUMING sample is sampled from targeted_sampler.reference
+    """
+
```

Comment:
Are we using targeted_sampler just to get an idea of target? 

Or is targeted sampler meant to have computed the cross terms in the covariance as well?

Reply:
targeted_sampler is used getting target_cov, but its use was mostly for using this method target_sampler.log_randomization_density() to evaluate the weights.

Reply:
Is targeted_sampler an instance of optimization_sampler?

Reply:
Yes. It gets called in the confidence interval method of optimization_sampler: https://github.com/jelena-markovic/Python-software/blob/refactor/selection/randomized/query.py#L1308